vendor/pip-pop/pip-diff

```diff
@@ -12,9 +12,24 @@ Options:
 """
 import os
 from docopt import docopt
-from pip.req import parse_requirements
-from pip.index import PackageFinder
-from pip._vendor.requests import session
+
```

Comment:
Can you say more about why these updates are required?

Reply:
The `pip-diff` and `pip-grep` tools  import internal modules from pip. In pip >= 10, internal modules were moved under `pip._internal`, breaking the imports. This PR uses `try...except ImportError` to handle both import paths.

Also, the interface of the `PackageFinder` class from one of the internal pip modules changed. This PR provides a wrapper function to allow creating objects of this type using the old interface.

Reply:
Thanks for the context 💯